lib/compat/wordpress-6.0/block-template-utils.php

```diff
@@ -26,21 +26,44 @@ function gutenberg_generate_block_templates_export_file() {
 	$filename = get_temp_dir() . 'edit-site-export-' . $obscura . '.zip';
 
 	$zip = new ZipArchive();
-	if ( true !== $zip->open( $filename, ZipArchive::CREATE ) ) {
+	if ( true !== $zip->open( $filename, ZipArchive::CREATE | ZipArchive::OVERWRITE ) ) {
 		return new WP_Error( 'unable_to_create_zip', __( 'Unable to open export file (archive) for writing.', 'gutenberg' ) );
 	}
 
-	$zip->addEmptyDir( 'theme' );
-	$zip->addEmptyDir( 'theme/templates' );
-	$zip->addEmptyDir( 'theme/parts' );
+	$theme_name = wp_get_theme()->get('TextDomain');
+
+	$zip->addEmptyDir( $theme_name );
+	$zip->addEmptyDir( $theme_name . '/templates' );
+	$zip->addEmptyDir( $theme_name . '/parts' );
```

Comment:
this could also be `block-templates` and `block-template-parts`, both are supported. Maybe `get_block_theme_folders()` can help here?